sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/FilterEstimation.scala

```diff
@@ -395,27 +395,28 @@ case class FilterEstimation(plan: Filter) extends Logging {
     // use [min, max] to filter the original hSet
     dataType match {
       case _: NumericType | BooleanType | DateType | TimestampType =>
-        val statsInterval =
-          ValueInterval(colStat.min, colStat.max, dataType).asInstanceOf[NumericValueInterval]
-        val validQuerySet = hSet.filter { v =>
-          v != null && statsInterval.contains(Literal(v, dataType))
-        }
+        if (colStat.min.isDefined && colStat.max.isDefined) {
```

Comment:
check `ndv == 0` at the beginning and return `Some(0.0`? then we don't have to make all these changes

Reply:
Yes, I have removes the bigger if, and implemented all three checks with one small if